doc/decisions/0_drafts/errors_plugins.md

```diff
@@ -0,0 +1,103 @@
+# Error Semantics
+
+## Problem
+
+While we have a classification of errors and warnings, it remains unclear when plugins actually should emit errors and warnings.
+
+## Constraints
+
+- Should not be contradicting to specified behavior in [storage plugin tutorial](/doc/tutorials/storage-plugins.md).
+
+## Assumptions
+
+- Users want a uniform behavior within Elektra, so plugins must behave uniformly.
+
+## Considered Alternatives
+
+### Free-for-all
+
+Every plugin author decides for themselves.
+
+Leads to inconsistent behaviour and hard to diagnose interaction issues.
+
+### Detailed ruleset with conformance tests
+
+Write a detailed formal ruleset and generic conformance tests for all plugins.
+
+Defining the ruleset, let alone writing the tests, for all possible errors in plugins is borderline impossible.
+
+### Simple guidelines
+
+Create simple guidelines for plugin authors and clearly document them.
+Rely on plugin authors adhering to the guidelines and testing their plugins themselves.
+Plugins that do not adhere to the guidelines can be marked down via `infos/status`.
+
+The basic guidelines will be (but may be extended):
+
+1. During `kdbGet` only emit warnings, unless the error is entirely unrecoverable:
+   Whatever other guidelines say, during `kdbGet` we plugins shall only emit warnings to allow the caller to fix issues via `kdbSet`.
+   Only if errors are entirely unrecoverable, should an error be emitted.
+   Some examples for either case:
+
+   1. File-based backend cannot parse local file:
+      Emit a warning.
+      We may still be able to overwrite the file in `kdbSet`.
+   2. File-based backend cannot access file, because path is invalid (e.g., on Windows `C:\foo\CON`)
+      Emit an error.
+      A `kdbSet` could recover, but the mountpoint definition needs to change.
+      To do this, a new `kdbOpen` is needed.
+
+2. Prefer errors to warnings in `kdbSet`:
+   Most validation detect either valid or invalid values.
+   Only in the rare case, where something is valid, but might not be correct should a warning be used.
+
+   For very common mistakes which are still technically valid configuration, both emitting an error and emitting a warning are valid options.
+   But in both cases, there should be a way to suppress or avoid the error/warning.
+
+3. Don't stop after the first error:
+   Additional errors, are automatically converted into warnings.
+   Report as many errors as possible in one go, so the user can fix as much as possible in one go.
+
+4. Specifications are exact:
+   Both the checked config must conform **exactly** to the specification.
+   Plugins should not ignore unknown values or values they don't understand.
+
+5. Reject unknowns:
+   To be on the safe side, always reject any unknowns that are direct at the plugin.
```

Comment:
"direct at"?

Reply:
Should be "directed at"

```suggestion
   To be on the safe side, always reject any unknowns that are directed at the plugin.
```

I needed a general term for "things the plugin should handle". Not all plugins are triggered by presence of metadata. Some (like storage plugins) try to handle everything, others get a parent from somewhere, etc.